nomad/core_sched.go

```diff
@@ -773,6 +791,82 @@ func (c *CoreScheduler) expiredOneTimeTokenGC(eval *structs.Evaluation) error {
 	return c.srv.RPC("ACL.ExpireOneTimeTokens", req, &structs.GenericResponse{})
 }
 
+// expiredACLTokenGC handles running the garbage collector for expired ACL
+// tokens. It can be used for both local and global tokens and includes
+// behaviour to account for periodic and user actioned garbage collection
+// invocations.
+func (c *CoreScheduler) expiredACLTokenGC(eval *structs.Evaluation, global bool) error {
+
+	expiryThreshold := c.getThreshold(eval, "expired_acl_token",
+		"acl_token_expiration_gc_threshold", c.srv.config.ACLTokenExpirationGCThreshold)
+
+	expiredIter, err := c.snap.ACLTokensByExpired(global)
+	if err != nil {
+		return err
+	}
+
+	var (
+		expiredAccessorIDs []string
+		num                int
+	)
+
+	// The memdb iterator contains all tokens which include an expiration time,
+	// however, as the caller, we do not know at which point in the array the
+	// tokens are no longer expired. This time therefore forms the basis at
+	// which we draw the line in the iteration loop and find the final expired
+	// token that is eligible for deletion.
+	now := time.Now().UTC()
+
+	for raw := expiredIter.Next(); raw != nil; raw = expiredIter.Next() {
+		token := raw.(*structs.ACLToken)
+
+		// The indexes mean if we come across an unexpired token, we can exit
+		// as we have found all currently expired tokens.
+		if !token.IsExpired(now) {
+			break
+		}
+
+		// Check whether the token is old enough to be deleted and skip if it
+		// is not.
```

Comment:
The phrasing of the comment is opposite of the code - would rather read something like, "Check if the token is recent enough to skip, otherwise we'll delete it"